### PR TITLE
[pkg/otlp] Add config for OTLP log ingestion

### PR DESCRIPTION
**Pull request by IbraheemA**
### What does this PR do?

Add configuration variables that will be used to enable log ingestion in the OTLP agent. These will be picked up by `map_provider.go` when the pipeline is being initialized, to create an exporter that handles logs.

Adding `logs:` to the `otlp_config:` in `datadog.yaml` will enable them.

[An RFC with more information can be found here.](https://docs.google.com/document/d/1nIcH_5YFFDCNzIy1gDL2OresVovrW2hbYfGY0v4Sp-A/edit)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/otlp/config_test.go [hunk 1]
```diff
@@ -480,6 +495,7 @@ func TestFromAgentConfigDebug(t *testing.T) {
 				TracePort:                5003,
 				MetricsEnabled:           true,
 				TracesEnabled:            true,
+				LogsEnabled:              true,
```
Comment (songy23) [inline pkg/otlp/config_test.go:498]: Please add a test on DD_OTLP_CONFIG_LOGS_ENABLED to TestFromEnvironmentVariables

File: releasenotes/notes/otlp-log-ingestion-config-6da2fca24efe1fad.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+features:
+  - |
+    Add config variables related to OTLP log ingestion
```
Comment (songy23) [inline releasenotes/notes/otlp-log-ingestion-config-6da2fca24efe1fad.yaml:3]:
```suggestion
    Add config variables and environment variable ``DD_OTLP_CONFIG_LOGS_ENABLED `` related to OTLP log ingestion
```

**Comment by songy23** [comment]
Note: QA instructions will be included in the follow-up PR with OTLP logs exporter implementation.

**Review by drichards-87 — APPROVED** [review]

File: pkg/otlp/config_test.go [hunk 3]
```diff
@@ -351,6 +361,45 @@ func TestFromEnvironmentVariables(t *testing.T) {
...
+				TracePort:                5003,
+				Metrics: map[string]interface{}{
+					"enabled":         true,
+					"tag_cardinality": "low",
+				},
+				Debug: map[string]interface{}{},
+			},
+		},
+		{
+			name: "logs disabled",
```
Comment (songy23) [inline pkg/otlp/config_test.go:374]: This test and the next one are the same, do you intend to test logs enabled?

File: pkg/util/intern/string.go [hunk 4]
```diff
@@ -102,6 +103,7 @@ func (s *StringInterner) GetString(k string) *StringValue {
 // The returned pointer will be the same for Get(v) and Get(v2)
 // if and only if v == v2. The returned pointer will also be the same
 // for a string with same contents as the byte slice.
+//
```
Comment (brycekahle) [inline pkg/util/intern/string.go:106]: This formatting change should probably not be in this PR since it has nothing to do with this file.

File: pkg/gohai/cpu/lscpu_linux_arm64.go [hunk 5]
```diff
@@ -19,7 +19,7 @@ type hwImpl struct {
 //
 // See from-lscpu-arm.py to regenerate this value.
 var hwVariant = map[uint64]hwImpl{
-	0x41: hwImpl{
+	0x41: {
```
Comment (songy23) [inline pkg/gohai/cpu/lscpu_linux_arm64.go:22]: Looks like unrelated changes?

File: pkg/logs/internal/pb/agent_logs_payload.pb.go [hunk 6]
```diff
@@ -2,13 +2,15 @@
 // source: agent_logs_payload.proto
 
 /*
-	Package pb is a generated protocol buffer package.
+Package pb is a generated protocol buffer package.
```
Comment (songy23) [inline pkg/logs/internal/pb/agent_logs_payload.pb.go:5]: Same

File: pkg/util/intern/string.go [hunk 7]
```diff
@@ -78,6 +78,7 @@ func NewStringInterner() *StringInterner {
 // The returned pointer will be the same for GetString(v) and GetString(v2)
 // if and only if v == v2. The returned pointer will also be the same
 // for a byte slice with same contents as the string.
+//
```
Comment (songy23) [inline pkg/util/intern/string.go:81]: Same

File: rtloader/test/helpers/helpers.go [hunk 8]
```diff
@@ -24,6 +24,7 @@ var (
 )
 
 // TestMemoryTracker is the method exposed to the RTLoader for memory tracking
+//
```
Comment (songy23) [inline rtloader/test/helpers/helpers.go:27]: ditto

**Review by songy23 — COMMENTED** [review]
Please avoid force push once reviewers begin reviewing, otherwise it's hard to see the incremental diffs.

**Review by songy23 — APPROVED** [review]
LGTM, please ping #agent-shared-components to get an owner review

File: releasenotes/notes/otlp-log-ingestion-config-6da2fca24efe1fad.yaml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+features:
+  - |
+    Add config variables and environment variable ``DD_OTLP_CONFIG_LOGS_ENABLED `` related to OTLP log ingestion
```
Comment (hush-hush) [inline releasenotes/notes/otlp-log-ingestion-config-6da2fca24efe1fad.yaml:3]: We should list the new config name instead of the env vars and explain it's behavior.

WDYT of:
```suggestion
    Adding new configuration setting ``otlp_config.logs.enabled`` to enable/disable logs support  in the OTLP ingest endpoint.
```

**Review by hush-hush — APPROVED** [review]
